From: Nell Hardcastle <nell@dev-nell.com>
Date: Thu, 29 May 2014 22:06:50 -0700
Subject: [PATCH] eMMC: Don't initialize partitions on RPMB flagged areas.
Origin: https://dev-nell.com/rpmb-emmc-errors-under-linux.html
Bug-Debian: https://bugs.debian.org/782038

Prevents a lot of pointless hanging at boot on some devices.
---
[bwh: An RPMB is unlikely to be useful on any platform supported by a Debian
kernel, so I think it's best to hide them for now.  Once we are confident
that userland will ignore them or the kernel does a better job of avoiding
hanging reads, we can drop this patch.]

 drivers/mmc/card/block.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -2258,7 +2258,7 @@ static int mmc_blk_alloc_parts(struct mm
 		return 0;
 
 	for (idx = 0; idx < card->nr_parts; idx++) {
-		if (card->part[idx].size) {
+		if (card->part[idx].size && !(card->part[idx].area_type & MMC_BLK_DATA_AREA_RPMB)) {
 			ret = mmc_blk_alloc_part(card, md,
 				card->part[idx].part_cfg,
 				card->part[idx].size >> 9,
